File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums)-1
        while low<=high:
            m = (low+high)//2
            print(m)
            if nums[m]>=target:
                high = m-1
            else:
                low = m+1
        return low
        

File: 35-search-insert-position/search-insert-position.cpp
class Solution {
public:
    int lb(vector<int>& nums, int target, int l, int h) {
        if (l>=h) return l;
        int m = round((l+h)/2);
        if (nums[m]>=target) return lb(nums, target, l, m);
        else return lb(nums, target, m+1, h);
    }
    int searchInsert(vector<int>& nums, int target) {
        return lb(nums, target, 0, nums.size());
    }
};